### ISSUE TITLE
Add caching

### ISSUE THREAD

**Issue by artlowel** [DSpace/dspace-angular#32]
This task depends on #31 

- Each Data service created in #31  should have a cache. 
- users of the data services (meaning components, other services, …) shouldn't need to know about the cache. They should just go on working the same way (with the same observables) as if the data came from the REST API.
- This cache should store its data in the ngrx store
- It should be implemented in such a way that most code can be shared between data services for different endpoints
- But it should be possible to determine separate time-to-live for each endpoint. For some endpoints it may be necessary to disable the cache altogether
- It should also be possible to combine partial objects:
	- e.g. I first get only a collection's id, name and handle from the REST API, for a certain page
	- next I go to another page and I get that same collection's id and shortDescription from the REST API, but not it's name or handle
	- the store should now contain the collection's name, handle and shortDescription

**Comment by artlowel** [comment]
Initial cache is done.

This ticket was left open because the code still lacks tests and documentation. This was moved to a separate issue: https://github.com/DSpace/dspace-angular/issues/130
